Lesson16/sinem16/script.js

```diff
@@ -255,100 +299,95 @@ console.log(
       to collect only the words that have a length >= minLength.
     - Log the resulting array.
 */
+console.log("Exp.15");
+
 function filterLongWords(words, minLength) {
   let longWords = [];
-  for (let i = 0; i < words.length; i++) {
-    const currentWord = words[i];
-    if (currentWord.length >= minLength) {
-      longWords.push(currentWord);
+  for (let char of words) {
+    if (char.length >= minLength) {
+      longWords.push(char);
     }
   }
-  return longWords;
+  console.log(longWords);
 }
 
-const fruitWords = ["apple", "banana", "cherry", "strawberry", "pear", "kiwi"];
-console.log(filterLongWords(fruitWords, 5));
-console.log(filterLongWords(fruitWords, 6));
+filterLongWords(["words", "sea", "like", "cat", "dog", "no", "at"], 3);
 
 /*
 16. Log Array Elements with Their Indices
     - Define a function `logElementsWithIndex(arr)` that loops through the array
       and logs "Index: i, Value: arr[i]" for each element.
 */
+console.log("Exp.16");
+
 function logElementsWithIndex(arr) {
-  for (let i = 0; i < arr.length; i++) {
-    console.log(`Index: ${i}, Value: ${arr[i]}`);
+  let index = [];
+  for (let key in arr) {
+    console.log("İndex: ", key, "Value: ", arr[key]);
   }
 }
-logElementsWithIndex(fruitWords);
+
+logElementsWithIndex([2, 4, 6, 8, 10, 12]);
 
 /*
 17. Find the Smallest Number in an Array
     - Define a function `findMin(numbers)` that loops through the array
       to find and return the smallest number.
     - Log the smallest number.
 */
+console.log("Exp.17");
+
 function findMin(numbers) {
-  let minNumber = numbers[0];
-  for (let i = 1; i < numbers.length; i++) {
-    if (numbers[i] < minNumber) {
-      minNumber = numbers[i];
+  minNum = numbers[0];
```

Comment:
```suggestion
  let minNum = numbers[0];
```

Reply:
If you forget to write let or const before a variable name, it becomes a global variable. This is dangerous because it can overwrite data in other parts of your application.